ui/src/components/FacetCard.js

```diff
@@ -10,17 +10,21 @@ class FacetCard extends Component {
     constructor(props) {
         super(props);
 
-        this.facetValues = this.props.facet.values;
-        this.totalCount = this.props.totalCount;
-
         this.state = {
-            selectedValues: []
+            selectedValues: [],
+            facetCount: this.sumCounts(this.props.facet.values),
         };
 
         this.onValueCheck = this.onValueCheck.bind(this);
         this.isUnselected = this.isUnselected.bind(this);
     }
 
+    componentWillReceiveProps(nextProps) {
```

Comment:
Overall this PR is pretty complicated. Constructor state initialization calls sumCounts() which calls isUnselected() which references state.

Here is what I believe is the idiomatic React way, which should be much simpler:

- Have FacetValue store isSelected. So FacetValue consists of name, count, isSelected.
- Render checkbox based on isSelected
- sumCounts() will be simpler. Note it will have special logic: if nothing is selected, sum everything.

FYI https://reactjs.org/blog/2018/03/29/react-v-16-3.html says not to use componentWillReceiveProps

Reply:
N/m, looks like you are using componentWillReceiveProps correctly. In future version of React, this will change to getDerivedStateFromProps